src/sentry/workflow_engine/detectors/error.py

```diff
@@ -0,0 +1,173 @@
+
+from sentry import analytics
+from sentry.eventstore.models import Event
+from sentry.integrations.tasks.kick_off_status_syncs import kick_off_status_syncs
+from sentry.issues import grouptype
+from sentry.metrics.middleware import MutableTags
+from sentry.models.activity import Activity
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
+from sentry.models.groupinbox import GroupInboxRemoveAction, remove_group_from_inbox
+from sentry.models.options.project_option import ProjectOption
+from sentry.models.project import Project
+from sentry.signals import issue_resolved
+from sentry.types.activity import ActivityType
+from sentry.workflow_engine.models import DataPacket
+from sentry.workflow_engine.models.data_condition import (
+    DetectorWorkflowCondition,
+    condition_registry,
+)
+from sentry.workflow_engine.processors.detector import DetectorEvaluationResult, DetectorHandler
+from sentry.workflow_engine.types import DetectorGroupKey
+
+T = TypeVar("T")
+
+
+class ErrorDetectorConfigType(StrEnum):
+    GROUPING = "ErrorGroupingCondition"
+    RESOLVE = "ErrorResolveCondition"
+
+
+class ErrorDetectorHandler(DetectorHandler[T]):
+    def evaluate(
+        self, data_packet: DataPacket[T]
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult]:
+        pass
+
+    def apply(self, type: ErrorDetectorConfigType, data: dict[str, Any]):
+        assert self.condition_group
+        config = self.condition_group.conditions.get(
+            type=type
+        )  # unused condition model for project options. can migrate later, but make sure it exists
+        assert config
+        condition = condition_registry.get(type)
+        return condition.apply(data)
+
+    def check_auto_resolve_project(self, project_id, options, cutoff):
+        if not options.get("sentry:resolve_age"):
+            # kill the option to avoid it coming up in the future
+            ProjectOption.objects.filter(
+                key__in=["sentry:_last_auto_resolve", "sentry:resolve_age"], project=project_id
+            ).delete()
+            return False
+
+        if int(options.get("sentry:_last_auto_resolve", 0)) > cutoff:
+            return False
+
+        return True
+
+
+Job = MutableMapping[str, Any]
+
+
+@condition_registry.register("ErrorGroupingCondition")
+@dataclass
+class ErrorGroupingCondition(DetectorWorkflowCondition):
+    data_dict = TypedDict(
+        "ErrorGroupingDict", {"event": Event, "job": Job, "metric_tags": MutableTags}
+    )
+
+    @classmethod
+    def apply(cls, data: dict[str, Any]):
+        data_dict = cls.data_dict(**data)
+
+        from sentry.event_manager import assign_event_to_group
+
+        # nothing to do with getting the DataCondition here...
+        return assign_event_to_group(**data_dict)
+
+
+@condition_registry.register("ErrorResolveCondition")
+@dataclass
+class ErrorResolveCondition(DetectorWorkflowCondition):
```

Comment:
This feels like quite a bit more than a condition, how can we simplify this down to just the logic of evaluating the group? Can we refactor any code to reuse code that's in the issue platform? It seems like we're doing a _lot_ of logic in here, how can we decompose this class?